### DISCUSSION TITLE
Filesystem Provider REMOTE - Add NFS

### DISCUSSION THREAD

**Discussion by marcschh** [music-assistant/support#1198] [Feature requests and ideas]
Hi,

currently it is possible to add SMB Shares but not NFS.
Then I installed nfs-utils in HA and connected my NFS-share in a subfolder of /media
the music-assistant-addon(docker) image seem missing nfs-utils as well as i is not able to view the content of the directory.
would be great if NFS could be added as well.

thanks
Marc

**Comment by compujunk** [comment]
Yes I would also like to see NFS back in MA as I don't normally use SMB on my network. I have temporarily set it up for MA now, but I want to get rid of it as soon as possible

Thanks René

**Comment by greymatter** [comment]
ill jump in here as well as I just installed MA and saw this was missing, all of my music is shared as both NFS and SMB however I've seen the performance of NFS to be much better over the years.  Would be a great addition!

**Comment by HaDe88** [comment]
I also just installed MA. I only have NFS shares and would love to see it supported so I can start using MA.

**Comment by compujunk** [comment]
You can make the NFS Mount in HA at Settings Storage and then add network storage. There you can then cheat for an NFS mount.
If it is created you can add it in MA as being Local Disk

**Reply by HaDe88** [reply]
Works like a charm! Thank you!

**Comment by meiser79** [comment]
Hi, which path do you use in MA? I tried /mnt/data/supervisor/share/Media (this is the NFS share from HA), but MA does not see this path.

**Reply by compujunk** [reply]
depends on how you mounted it in HA 

<img width="606" height="830" alt="image" src="https://github.com/user-attachments/assets/adf15d72-7949-41c7-a814-7c3804ffbb01" />

If you have it as above then you have to do it per directory like this

<img width="1353" height="544" alt="image" src="https://github.com/user-attachments/assets/4a2d1441-082b-4bd4-b358-e604fd03e8e4" />

**Comment by OzGav** [comment]
NFS filesystem provider will be available from 2.8.0b20

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
